### PR TITLE
[WIP] - Make deployer controller to respect 3 laws of controllers

### PR DESCRIPTION
**Pull request by tnozicka**
We could potentially delete wrong deployer pods if the same DC is deleted and recreated.

/cc @mfojtik 

(needs tests)

### CODE REVIEW

File: pkg/apps/controller/deployer/deployer_controller.go [hunk 1]
```diff
@@ -108,6 +108,32 @@ func (c *DeploymentController) handle(deployment *v1.ReplicationController, will
...
 	deployer, deployerErr := c.podLister.Pods(deployment.Namespace).Get(deployerPodName)
+
+	selector := deployutil.DeployerPodSelector(deployment.Name)
+	cm := kcontroller.NewPodControllerRefManager(c.pc, deployment, selector, deployutil.RCControllerRefKind, c.canAdoptFunc(deployment))
+	_, err := cm.ClaimPods([]*v1.Pod{deployer})
+	if err != nil {
+		return err
+	}
+
+	{
```
Comment (mfojtik) [inline pkg/apps/controller/deployer/deployer_controller.go]: why is this block?

File: pkg/apps/controller/deployer/deployer_controller.go [hunk 2]
```diff
@@ -375,6 +389,7 @@ func (c *DeploymentController) makeDeployerPod(deployment *v1.ReplicationControl
 
 	gracePeriod := int64(10)
 
+	t := true
```
Comment (mfojtik) [inline pkg/apps/controller/deployer/deployer_controller.go]: I wish there is BoolPtr() helper ;-)

File: pkg/apps/controller/deployer/deployer_controller.go (hunk 1 above)
Comment (tnozicka) [reply pkg/apps/controller/deployer/deployer_controller.go]: to scope variables and not to leak names since this is to be removed after the interim

File: pkg/apps/controller/deployer/deployer_controller.go (hunk 2 above)
Comment (soltysh) [reply pkg/apps/controller/deployer/deployer_controller.go]: Please create one and don't use `t` as a variable name, it's hard to follow what's it's about.

File: pkg/apps/strategy/support/lifecycle.go [hunk 3]
```diff
@@ -384,6 +384,7 @@ func makeHookPod(hook *deployapi.LifecycleHook, rc *kapi.ReplicationController,
 	podSecurityContextCopy := rc.Spec.Template.Spec.SecurityContext.DeepCopy()
 	securityContextCopy := baseContainer.SecurityContext.DeepCopy()
 
+	t := true
```
Comment (soltysh) [inline pkg/apps/strategy/support/lifecycle.go]: Same here.
